mlflow/genai/judges/instructions_judge/__init__.py

```diff
@@ -0,0 +1,271 @@
+    def model(self) -> str:
+        """Get the model for this judge."""
+        return self._model
+
+    @property
+    def template_variables(self) -> set[str]:
+        """Get the template variables from the instructions."""
+        return self._instructions_prompt.variables
+
+    @property
+    def description(self) -> str:
+        """Get the description of this judge."""
+        header = f"Instructions-based judge: {self.name}"
+        return f"{header}\n\nInstructions:\n-------------\n\n{self._instructions}"
+
+    def __call__(
+        self,
+        *,
+        inputs: dict[str, Any] | None = None,
+        outputs: dict[str, Any] | None = None,
+        expectations: dict[str, Any] | None = None,
+        trace: Trace | None = None,
+    ) -> Any:
+        """
+        Evaluate the provided data using the judge's instructions.
+
+        NOTE: Evaluation supports either inputs, outputs, and/or expectations
+        or trace as input. A trace cannot be used with any other fields.
+
+        Args:
+            inputs: Input dictionary to evaluate. Cannot be used with 'trace'.
+            outputs: Output dictionary to evaluate. Cannot be used with 'trace'.
+            expectations: Expected outcomes or ground truth.
+            trace: Trace object for evaluation.
+
+        Returns:
+            Evaluation results
+
+        """
+        # TODO: update this to perform a mutual exclusion validiation check if both
+        # trace and inputs / outputs / expectations are submitted once trace template
+        # substitution is implemented
+        if trace is not None:
+            raise MlflowException(
+                "Trace evaluation is not supported in this version. "
+                "Please use 'inputs' and 'outputs' for evaluation.",
+                error_code=INVALID_PARAMETER_VALUE,
+            )
+        if inputs is not None or outputs is not None:
+            self._validate_call_args_contain_template_fields(inputs, outputs, expectations)
+
+            template_values = {}
+            if inputs is not None:
+                template_values.update(inputs)
+            if outputs is not None:
+                template_values.update(outputs)
+            if expectations is not None:
+                if self._TEMPLATE_VARIABLE_EXPECTATIONS in self.template_variables:
+                    template_values[self._TEMPLATE_VARIABLE_EXPECTATIONS] = json.dumps(
+                        expectations, default=str, indent=2
+                    )
+                template_values.update(expectations)
+
+            formatted_prompt = format_prompt(self._instructions, **template_values)
+
+            return invoke_judge_model(
+                model_uri=self._model,
+                prompt=formatted_prompt,
+                assessment_name=self.name,
+            )
+        # TODO: implement trace template parsing
+        raise MlflowException(
+            "Must specify either 'inputs'/'outputs' or 'trace' for evaluation.",
```

Comment:
```suggestion
            "Must specify either 'inputs' or 'outputs' for evaluation.",
```

(We'll add support for `trace` in a followup PR)